api/envoy/extensions/filters/http/json_to_metadata/v3/json_to_metadata.proto

```diff
@@ -0,0 +1,121 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.json_to_metadata.v3;
+
+import "google/protobuf/struct.proto";
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.json_to_metadata.v3";
+option java_outer_classname = "JsonToMetadataProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/json_to_metadata/v3;json_to_metadatav3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Json-To-Metadata Filter]
+//
+// The configuration for transforming json body into metadata. This is useful
+// for matching load balancer subsets, logging, etc.
+//
+// Json to Metadata :ref:`configuration overview <config_http_filters_json_to_metadata>`.
+// [#extension: envoy.filters.http.json_to_metadata]
+
+message JsonToMetadata {
+  enum ValueType {
+    // The value is a serialized `protobuf.Value
+    // <https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/struct.proto#L62>`_.
+    PROTOBUF_VALUE = 0;
+
+    STRING = 1;
+
+    NUMBER = 2;
+  }
+
+  // [#next-free-field: 6]
+  message KeyValuePair {
+    // The namespace — if this is empty, the filter's namespace will be used.
+    string metadata_namespace = 1;
+
+    // The key to use within the namespace.
+    string key = 2 [(validate.rules).string = {min_len: 1}];
+
+    oneof value_type {
+      // The value to pair with the given key.
+      //
+      // When used for on_present case, if value is non-empty it'll be used instead
+      // of the attribute value. If both are empty, the attribute value is used as-is.
```

Comment:
What does the term "attribute" refer to here?  Does it mean the value of the JSON key?  If so, I suggest using a phrase like "the value of the JSON key" instead.

Same thing throughout.

Reply:
thanks for the suggestion! this wording bothers me a lot

Reply:
most of the cases "attribute" means the key-value pair of the desired target.
However "attribute value" should be "the value of the JSON key"